tests/e2e/common-actions/browser-actions.ts

```diff
@@ -39,17 +40,62 @@ export class BrowserActions {
             ? await t.expect(browserPage.tooltip.textContent).contains(expectedText, `"${expectedText}" Text is incorrect in tooltip`)
             : await t.expect(browserPage.tooltip.textContent).notContains(expectedText, `Tooltip still contains text "${expectedText}"`);
     }
+
     /**
      * Verify that the new key is displayed at the top of the list of keys and opened and pre-selected in List view
-     * */
+     * @param keyName Key name
+     */
     async verifyKeyDisplayedTopAndOpened(keyName: string): Promise<void> {
         await t.expect(Selector('[aria-rowindex="1"]').withText(keyName).visible).ok(`element with ${keyName} is not visible in the top of list`);
         await t.expect(browserPage.keyNameFormDetails.withText(keyName).visible).ok(`element with ${keyName} is not opened`);
     }
+
     /**
      * Verify that the new key is not displayed at the top of the list of keys and opened and pre-selected in List view
-     * */
+     * @param keyName Key name
+     */
     async verifyKeyIsNotDisplayedTop(keyName: string): Promise<void> {
         await t.expect(Selector('[aria-rowindex="1"]').withText(keyName).exists).notOk(`element with ${keyName} is not visible in the top of list`);
     }
+
+    /**
+     * Check tree view structure
+     * @folders name of folders for tree view build
+     * @delimiter string with delimiter value
+     * @commonKeyFolder flag if not patterned keys will be displayed
+     */
+    async checkTreeViewFoldersStructure(folders: string[][], delimiter: string, commonKeyFolder: boolean): Promise<void> {
+        // Verify that all keys that are not inside of tree view doesn't contain delimiter
+        if (commonKeyFolder) {
+            const notPatternedKeys = Selector('[data-testid^="badge"]').parent('[data-testid^="node-item_"]');
+            const notPatternedKeysNumber = await notPatternedKeys.count;
+            for (let i = 0; i < notPatternedKeysNumber; i++) {
+                await t.expect(notPatternedKeys.nth(i).withText(delimiter).exists).notOk('Not contained delimiter keys');
+            }
+        }
+        // Verify that every level of tree view is clickable
+        const foldersNumber = folders.length;
+        for (let i = 0; i < foldersNumber; i++) {
+            const innerFoldersNumber = folders[i].length;
+            const array: string[] = [];
+            for (let j = 0; j < innerFoldersNumber; j++) {
+                if (j === 0) {
+                    const folderSelector = `[data-testid="node-item_${folders[i][j]}${delimiter}"]`;
+                    array.push(folderSelector);
+                    await t.click(Selector(folderSelector));
+                }
+                else {
+                    const lastSelector = array[array.length - 1].substring(0, array[array.length - 1].length - 2);
```

Comment:
could you please clarify it for me?

Reply:
the purpose is to remove the last two characters from the last element in the array